### PR TITLE
Replace predicate functions with helper functions in Policy module

### PR DESCRIPTION
**Pull request by leonardodipace**
## Summary
- Removes support for raw predicate functions (`(v) => v === "draft"`) in policy conditions
- Introduces typed helper functions as the only way to express conditions beyond exact value matching
- Each helper carries operator and value metadata on its type, enabling future adapter support (e.g. SQL WHERE clause generation)

## Summary by CodeRabbit

* **New Features**
  * Adds a formal set of composable condition helpers (comparisons, logic combinators, string/array checks, presence checks) and exposes them via the policy API for building rules.

* **Documentation**
  * Policy docs and README updated with helper-based examples and guidance showing permission- and helper-driven conditions.

* **Tests**
  * Large new unit and integration test suite validating each helper, metadata propagation, edge cases, and integration with policy evaluation.

### CODE REVIEW

File: src/lib/policy/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+  return {
+    [_brand]: fn,
+    operator: "has",
+    value: items,
+    fn,
+  };
+};
+
+export const hasAny = <V>(items: V[]): ConditionHelper<V[]> => {
```
Comment (arci-arci) [inline src/lib/policy/helpers.ts]: in my opinion, `hasAny` should also accept a single item like in the `has` function

**Review by arci-arci — CHANGES_REQUESTED** [review]

File: src/lib/policy/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+  };
+};
+
+export const gt = (
+  value: string | number | Date,
+): ConditionHelper<string | number | Date> => {
+  const fn = (actual: string | number | Date) => actual > value;
+
+  return { [_brand]: fn, __isConditionHelper: true, operator: "gt", value, fn };
+};
```
[the remark this answers is not in the document]
Comment (leonardodipace) [reply src/lib/policy/helpers.ts]: The generic approach with a type parameter extending string | number | Date causes TypeScript to infer literal types at call sites - gt(5) resolves to ConditionHelper<5>, which is incompatible with ConditionHelper<number> in policy conditions. The overload signatures explicitly widen to the base types (string/number/Date), preserving assignability while still providing per-type specificity at the call site.

File: src/lib/policy/helpers.ts (hunk 1 above)
Comment (leonardodipace) [reply src/lib/policy/helpers.ts]: Done

**Review by arci-arci — APPROVED** [review]
all good for me
